src/Nancy/HttpLink.cs

```diff
@@ -0,0 +1,283 @@
+        /// <exception cref="System.ArgumentNullException"><paramref name="targetUri" /> or <paramref name="relation" /></exception>
+        public HttpLink(string targetUri, string relation, string type)
+            : this(ParseUri(targetUri), relation, type, string.Empty)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="HttpLink" /> class.
+        /// </summary>
+        /// <param name="targetUri">The target URI of the link.</param>
+        /// <param name="relation">The relation that identifies the semantics of the link.</param>
+        /// <param name="type">The optional type parameter is a hint indicating what the media type of the result of dereferencing the link should be. Note that this is only a hint; for example, it does not override the HTTP Content-Type header of a HTTP response obtained by actually following the link.</param>
+        /// <param name="title">The optional title parameter is used to label the destination of a link such that it can be used as a human-readable identifier (e.g., a menu entry) in the language indicated by the HTTP Content-Language header (if present).</param>
+        /// <exception cref="System.ArgumentNullException"><paramref name="targetUri" /> or <paramref name="relation" /></exception>
+        public HttpLink(string targetUri, string relation, string type, string title)
+            : this(ParseUri(targetUri), relation, type, title)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="HttpLink" /> class.
+        /// </summary>
+        /// <param name="targetUri">The target URI of the link.</param>
+        /// <param name="relation">The relation that identifies the semantics of the link.</param>
+        /// <param name="type">The optional type parameter is a hint indicating what the media type of the result of dereferencing the link should be. Note that this is only a hint; for example, it does not override the HTTP Content-Type header of a HTTP response obtained by actually following the link.</param>
+        /// <param name="title">The optional title parameter is used to label the destination of a link such that it can be used as a human-readable identifier (e.g., a menu entry) in the language indicated by the HTTP Content-Language header (if present).</param>
+        /// <exception cref="System.ArgumentNullException"><paramref name="targetUri" /> or <paramref name="relation" /></exception>
+        public HttpLink(Uri targetUri, string relation, string type, string title)
+        {
+            if (targetUri == null)
+            {
+                throw new ArgumentNullException("targetUri");
+            }
+
+            if (relation == null)
+            {
+                throw new ArgumentNullException("relation");
+            }
+
+            this.parameters = new Dictionary<string, object>
```

Comment:
> So, the most "correct" implementation would probably be something like IEnumerable<KeyValuePair<string, string>>

Are you saying that ideally you'd like a `MultiValueDictionary<string, string>`?

> the more pragmatic and usable thing would be to initialize a `Dictionary<string, string>` with `StringComparer.InvariantCultureIgnoreCase`.

Why `StringComparer.InvariantCultureIgnoreCase`? Wouldn't `StringComparer.OrdinalIgnoreCase` be "safer"?